### DISCUSSION TITLE
Getting a Codeium API Key

### DISCUSSION THREAD

**Discussion by necromeo** [leona/helix-gpt#60] [General]
First, this is a great project!

Somehow, I wasn't very keen on using the public Codeium API key. If anyone else is in the same position, here are the steps to getting your very own API key, provided you have a Codeium account.

1. Navigate to `https://www.codeium.com/profile?response_type=token&redirect_uri=vim-show-auth-token&state=a&scope=openid%20profile%20email&redirect_parameters_type=query`. This will provide you an auth token.
2. Once you have the token. Send a `POST` request to the following URL `https://api.codeium.com/register_user/` with this body
    ```json
    {
        "firebase_id_token": "<the-auth-token-from-step-one>"
    }
    ```
3. The response should look like this
    ```json
    {
    "api_key": "<the-api-key>",
    "name": "<your-codeium-username>"
    }
    ```
4. Voilà!

**Comment by testy-cool** [comment]
In case it helps anyone, open a terminal and run:
```
curl -X POST https://api.codeium.com/register_user/ -H 'Content-Type: application/json' -d '{"firebase_id_token": "YOUR_AUTH_TOKEN_HERE"}'
```
The resposne looks like this:
```
{"api_key":"sk-ws-abcdef...", "name":"Your Name"}
```
